api/src/wfl/module/covid.clj

```diff
@@ -226,21 +197,35 @@
         (set/rename-keys (set/map-invert tdr-source-serialized-fields)))
     (throw (ex-info "source_items is not an integer" details))))
 
-(defn ^:private create-tdr-source [tx id request]
-  (let [create  "CREATE TABLE %s OF TerraDataRepoSourceDetails (PRIMARY KEY (id))"
-        details (format "TerraDataRepoSourceDetails_%09d" id)
-        _       (jdbc/execute! tx [(format create details)])
-        items   (-> (select-keys request (keys tdr-source-serialized-fields))
-                    (set/rename-keys tdr-source-serialized-fields)
-                    (assoc :details details)
-                    (->> (jdbc/insert! tx tdr-source-table)))]
-    [tdr-source-type (-> items first :id)]))
+(defn ^:private peek-tdr-source-queue
+  "Get first unconsumed snapshot in DETAILS table."
+  [{:keys [details] :as _source}]
+  (let [query "SELECT *
+               FROM %s
+               WHERE consumed IS NULL
+               AND snapshot_id IS NOT NULL
```

Comment:
Flagging for @rexwangcc -- to be peekable, a member of the source queue (record in the `TerraDataRepoSourceDetails` instance) should have a defined `snapshot_id`.

From mobbing today, I'll rely on `snapshot_id` being defined for eligible records within your `update-tdr-source` loop.  Thank you!